src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -171,43 +176,76 @@ private IList<Message> ParsePayload(byte[] payload)
         private async Task SendMessages(Uri sendUrl, CancellationToken cancellationToken)
         {
             _logger.LogInformation("Starting the send loop");
-
-            TaskCompletionSource<object> sendTcs = null;
+            IList<SendMessage> messages = null;
             try
             {
                 while (await _application.Input.WaitToReadAsync(cancellationToken))
                 {
+                    // Grab as many messages as we can from the channel
+                    messages = new List<SendMessage>();
                     while (!cancellationToken.IsCancellationRequested && _application.Input.TryRead(out SendMessage message))
                     {
-                        sendTcs = message.SendResult;
+                        messages.Add(message);
+                    }
+
+                    if (messages.Count > 0)
```

Comment:
Yeah, I guess I just don't want to introduce an assert where one isn't needed. It's benign for the Task to complete and no messages to be available, since we'll just restart the loop.